runtime/src/main/java/org/corfudb/runtime/collections/vavr/HashArrayMappedTrie.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright 2023 Vavr, https://vavr.io
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ */
+
+package org.corfudb.runtime.collections.vavr;
+
+import io.vavr.Tuple2;
+import io.vavr.collection.Iterator;
+import io.vavr.control.Option;
+
+public interface HashArrayMappedTrie<K, V> extends Iterable<Tuple2<K, V>> {
```

Comment:
I think we can remove the dependence on the VAVR Iterator object and just use Java's since we convert between them through TupleIterableWrapper anyway. It seems like the dependence on VAVR can be removed entirely.

Reply:
Can we also add Javadocs for the newly added code and update the names for var1, var2 and the like?

Reply:
Yes, removing VAVR dependency involves a lot of changes and I was planning to have the change as a follow-up PR.

Reply:
Removed VAVR dependency in this PR itself.